## /bin/sh -e simulate dpatch for prepend_chlog.py
## 40_initgroups.dpatch by Kilian Krause <kk@verfaction.de>
## DP: does initialize groups of asterisk user if no -G is given.
## -- applied upstream for 1.2

Index: asterisk-1.2.24.dfsg+pf.0.1~svn1329/asterisk.c
===================================================================
--- asterisk-1.2.24.dfsg+pf.0.1~svn1329.orig/asterisk.c	2007-08-08 14:30:23.000000000 +0200
+++ asterisk-1.2.24.dfsg+pf.0.1~svn1329/asterisk.c	2007-08-08 14:30:25.000000000 +0200
@@ -2243,6 +2243,10 @@
 				exit(1);
 			}
 		}
+		if (!rungroup && initgroups(runuser, pw->pw_gid)) {
+			ast_log(LOG_WARNING, "Unable to initialize supplementary group list for %s\n", runuser);
+			exit(1);
+		}
 		if (setuid(pw->pw_uid)) {
 			ast_log(LOG_WARNING, "Unable to setuid to %d (%s)\n", (int)pw->pw_uid, runuser);
 			exit(1);
